lib/Memecoin.ts

```diff
@@ -466,39 +481,67 @@ export class MemecoinSDK {
     return BigInt(result)
   }
 
-  private async estimateSwapCoin(params: EstimateSwapCoinParams): Promise<bigint> {
-    const { fromToken, toToken, amountIn } = params
+  async estimateSwap(params: EstimateSwapParams): Promise<EstimateSwapResult> {
+    const [tokenInMemecoin, tokenOutMemecoin] = await Promise.all([
+      this.getCoin(params.tokenIn),
+      this.getCoin(params.tokenOut)
+    ])
 
-    const response = await fetch(`${this.apiBaseUrl}/api/trades/estimate-swap`, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        fromToken,
-        toToken,
-        amountIn: amountIn.toString()
-      })
-    })
+    const tokenInData = tokenInMemecoin
+      ? this.determinePoolOfMemecoin(tokenInMemecoin)
+      : await this.findTokenWETHPool(params.tokenIn)
+    const tokenOutData = tokenOutMemecoin
+      ? this.determinePoolOfMemecoin(tokenOutMemecoin)
+      : await this.findTokenWETHPool(params.tokenOut)
 
-    const result = await response.json()
+    const tokenInPoolType = tokenInData.poolType
+    const feeIn = tokenInData.poolFee
+    const tokenOutPoolType = tokenOutData.poolType
+    const feeOut = tokenOutData.poolFee
 
-    if (!isValidBigIntString(result)) {
-      throw new Error('Invalid response format')
+    const walletClient = this.walletClient
+    const account = walletClient.account
+    if (isNull(account)) {
+      throw new Error('No account found')
```

Comment:
wait, why do we need this? You're making the estimate function require a private key or account. We can estimate without a private key. we can pass it